### PR TITLE
[DependencyInjection] Fix nested env var with resolve processor

### LINKED ISSUE #44756: [DI] `resolve:` env processor can override real envs
### Description

See https://github.com/ro0NL/sf-env (6.1)

An issue occurs when using `%env(resolve:ENV)%`, when `ENV` contains a nested `%env(INNER)%`. The nested env is loaded from parameters, not real envs.

So given

```
TEST_ENV_INNER='im defined in .env'
TEST_ENV="pre [%env(TEST_ENV_INNER)%] post"
```

```yaml
parameters:
    env(TEST_ENV_INNER): 'im defined in services.yaml'

services:
    _defaults:
        bind:
            'string $testEnvValue': '%env(resolve:TEST_ENV)%'
```

Then running

```
$ bin/console test
^ "pre [im defined in services.yaml] post"
```

Expected is `pre [im defined in .env] post`

Now running `TEST_ENV_INNER=real bin/console test` produces the same output, whereas `pre [real] post` is expected

We found a hackish workaround using `env(INNER): '%env(INNER)%'`, but this should probably happen out-of-the-box ... as this is a circular reference IMHO.

cc @nicolas-grekas

### Example

_No response_

### PR DESCRIPTION
**Pull request by Laulibrius**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #44756
| License       | MIT
| Doc PR        | 

This PR fixes the issue raised by @ro0NL. (see Description)
I added some tests but i'm not confident about how to check with real environment variables.

This is my first PR, i hope i'm not doing it wrong.

### CODE REVIEW

**Comment by ro0NL** [comment]
IMHO this is correct behavior. It took some time to understand why the workaround was needed, when reviewing a PR on our side.

Im not sure about changing the behavior in a patch release. 6.1 + changelog works for us :+1:

File: src/Symfony/Component/DependencyInjection/EnvVarProcessor.php [hunk 1]
```diff
@@ -269,11 +269,17 @@ public function getEnv($prefix, $name, \Closure $getEnv)
...
 
         if ('resolve' === $prefix) {
-            return preg_replace_callback('/%%|%([^%\s]+)%/', function ($match) use ($name) {
+            return preg_replace_callback('/%%|%([^%\s]+)%/', function ($match) use ($name, $getEnv) {
                 if (!isset($match[1])) {
                     return '%';
                 }
-                $value = $this->container->getParameter($match[1]);
+
+                if (preg_match('/^env\(([^)]+)\)$/', $match[1], $submatch)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/EnvVarProcessor.php]: please don't use a regexp here, see how `EnvPlaceholderParameterBag` does it

**Comment by Laulibrius** [comment]
> Im not sure about changing the behavior in a patch release. 6.1 + changelog works for us 👍

I first did it on 6.1 but then you said it's a bug in your opinion so i followed the rules and switch to 4.4 to make this PR 😁 
I can switch back to 6.1 but i'm not sure what to write in the changelog... 🤔

**Review by nicolas-grekas — APPROVED** [review]
Works for me as a bugfix, thanks for the PR.

**Review by fancyweb — APPROVED** [review]
That's some nice tests :clap:

**Review by GromNaN — APPROVED** [review]

**Comment by GromNaN** [comment]
Thanks for fixing this bug Laurent.

**Pull request merged**